master/buildbot/test/integration/test_stop_trigger.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+
+from buildbot.test.util.integration import RunMasterBase
+from buildbot.status.results import CANCELLED
+
+# This integration test creates a master and slave environment,
+# with two builders and a trigger step linking them. the triggered build never ends
+# so that we can reliabily stop it recursively
+
+
+# master configurations
+
+def triggerRunsForever():
+    c = {}
+    from buildbot.config import BuilderConfig
+    from buildbot.process.factory import BuildFactory
+    from buildbot.plugins import steps, schedulers
+
+    c['schedulers'] = [
+        schedulers.Triggerable(
+            name="trigsched",
+            builderNames=["triggered"]),
+        schedulers.AnyBranchScheduler(
+            name="sched",
+            builderNames=["main"])]
+
+    f = BuildFactory()
+    f.addStep(steps.Trigger(schedulerNames=['trigsched'],
+                            waitForFinish=True,
+                            updateSourceStamp=True))
+    f.addStep(steps.ShellCommand(command='echo world'))
+
+    f2 = BuildFactory()
+    f2.addStep(steps.ShellCommand(command="\n".join(['while :',
+                                                     'do',
+                                                     ' echo "sleeping";',
+                                                     ' sleep 1;'
+                                                     'done'])))
+    c['builders'] = [
+        BuilderConfig(name="main",
+                      slavenames=["local1"],
+                      factory=f),
+        BuilderConfig(name="triggered",
+                      slavenames=["local1"],
+                      factory=f2)]
+    return c
+
+
+def triggeredBuildIsNotCreated():
+    c = {}
```

Comment:
you can still factorize those two functions. only 2 lines are different if I read correctly